analyzer/src/environment/variables.rs

```diff
@@ -31,7 +31,7 @@ pub struct Variables {
 
     /// Relations with external variables.
     /// The key is the variable Names, where value is the relation with another external environment's [Locals]
-    externals: HashMap<Name, RelationId>,
+    externals: IndexMap<Name, RelationId>,
```

Comment:
the compiler iterates over externals, we need a fixed order because otherwise the same source code would have different indexes for captured references

Reply:
As the captured list is recreated later by the compiler, this is not an issue.